kubetest/main.go

```diff
@@ -368,9 +370,23 @@ func findVersion() string {
 	return "unknown" // Sad trombone
 }
 
+// maybeMergeMetadata will add new keyvals into the map; quietly eats errors.
+func maybeMergeJSON(meta map[string]string, path string) {
+	if data, err := ioutil.ReadFile(path); err == nil {
+		json.Unmarshal(data, &meta)
+	}
+}
+
 // Write metadata.json, including version and env arg data.
-func writeMetadata(path string) error {
+func writeMetadata(o *options) error {
+	path := o.dump
 	m := make(map[string]string)
+
+	// Look for any sources of metadata and load 'em
+	for _, f := range strings.Split(o.metadataSources, ",") {
+		maybeMergeJSON(m, filepath.Join(path, f))
```

Comment:
You mean, add a prefix to each key read in from each file in `metadataSources` ? Seemed more flexible not to do that, at the risk of some possible confusion ...

I added some unit tests :)